specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx

```diff
@@ -288,7 +288,12 @@ export function InteractionDialog({
                 >
                   {interactionsText.addUnassociated()}
                 </Button.Info>
-              ) : model.name === 'Loan' || action.model.name === 'Loan' ? (
+              ) : (model.name === 'Loan' || action.model.name === 'Loan') &&
+                !(
+                  state.type === 'MissingState' &&
+                  prepsData?.length !== 0 &&
+                  prepsData
```

Comment:
can you double-check your code to see what does `&& prepsData` do in this code snippet under different possible values of `prepsData` to see if it is necessary